### DISCUSSION TITLE
GRPC is not being enabled in Wildfly 26.1.2.Final

### DISCUSSION THREAD

**Discussion by bantunes82** [wildfly-extras/wildfly-grpc-feature-pack#58] [Q&A]
Hi Guys

Do you know or have any documentation available to configure the GRPC server part in Wildfly version 26.1.2.Final ?
I followed this one https://github.com/wildfly-extras/wildfly-grpc-feature-pack/blob/main/docs/guide/index.adoc, but it didn't work.
 
I have tried this hello world bellow, but the GRPC port was not enabled in the log and the GRPC service was not registered as well.
It looks like the GPRC submodule is not being enabled, just to let you know we are not using the maven plugin "wildfly-maven-plugin" like you are using in yours helloworld example.

To enable this GRPC subsystem in Wildfly version 26.1.2.Final, do you know if I have to enable the subsystem via pre-commands.cli ?

This is java GRPC service that I added:
```java
@GrpcService
public class GreeterServiceImpl extends GreeterGrpc.GreeterImplBase {

    @Override
    public void sayHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
        String name = request.getName();
        String message = "Hello " + name;
        responseObserver.onNext(HelloReply.newBuilder().setMessage(message).build());
        responseObserver.onCompleted();
    }
}
```

These are the topics that I added in the pom.file

**dependency part**
```xml
<dependency>
	<groupId>org.wildfly.extras.grpc</groupId>
	<artifactId>wildfly-grpc-examples-helloworld-proto</artifactId>
	<version>0.0.3.Final</version>
</dependency>
<dependency>
	<groupId>org.wildfly.extras.grpc</groupId>
	<artifactId>wildfly-grpc-api</artifactId>
	<version>0.0.3.Final</version>
</dependency>
```

**plugin part**
```xml
<plugin>
	<groupId>org.apache.maven.plugins</groupId>
	<artifactId>maven-dependency-plugin</artifactId>
	<executions>
		<execution>
			<id>unpack-proto</id>
			<phase>initialize</phase>
			<goals>
				<goal>unpack</goal>
			</goals>
			<configuration>
				<artifactItems>
					<artifactItem>
						<groupId>org.wildfly.extras.grpc</groupId>
						<artifactId>wildfly-grpc-examples-helloworld-proto</artifactId>
						<version>0.0.3.Final</version>
					</artifactItem>
					</artifactItems>
						<outputDirectory>${project.build.directory}/proto</outputDirectory>
					</configuration>
		</execution>
	</executions>
</plugin>
<plugin>
	<groupId>org.xolstice.maven.plugins</groupId>
	<artifactId>protobuf-maven-plugin</artifactId>
	<executions>
		<execution>
			<id>compile-proto</id>
			<goals>
				<goal>compile</goal>
				<goal>compile-custom</goal>
			</goals>
			<configuration>
				<protoSourceRoot>${project.build.directory}/proto</protoSourceRoot>
			</configuration>
			</execution>
		</executions>
</plugin>
```

**Comment by ronsigal** [comment]
Hi @bantunes82,

Running `mvn install` in the root directory of wildfly-grpc-feature-pack will build a copy of WildFly in build/target/wildfly-27.0.1.Final-grpc-0.0.4-SNAPSHOTwildfly-27.0.1.Final-grpc-0.0.4-SNAPSHOT. 

Alternatively, you can create a WildFly with the grpc subsystem using galleon.sh on the command line. I just build one like this
>galleon.sh install wildfly-preview:current --dir=wildfly
galleon.sh install org.wildfly.extras.grpc:wildfly-grpc-feature-pack:0.0.4-SNAPSHOT --layers=grpc --dir=wildfly

Also, I built a WildFly based on 26.1.2.Final

>galleon.sh install wildfly-preview:current#26.1.2.Final --dir=my-wildfly

but it doesn't have an io/grpc module, so that might not work out without some fiddling around.

I copied wildfly-grpc-examples-helloworld-service-0.0.4-SNAPSHOT.jar into the deployments directory and saw

>18:00:54,583 INFO  [org.wildfly.extension.grpc] (ServerService Thread Pool -- 82) WFLYGRPC0003: gRPC server for wildfly-grpc-examples-helloworld-service-0.0.4-SNAPSHOT.jar listening on localhost:9555

Then I ran

>mvn exec:java -P examples -pl examples/helloworld/client -Dexec.args="Bob none"

and it worked. Note that the client needs a second argument now to determine what kind of SSL connection to create. I still have to clean up the examples a bit.

Does that help?

When you say, "I followed this one https://github.com/wildfly-extras/wildfly-grpc-feature-pack/blob/main/docs/guide/index.adoc, but it didn't work.", what part didn't work?
